From cd210f1210700a9917412cb6d312eb029236f1ae Mon Sep 17 00:00:00 2001
From: Michael Marineau <michael.marineau@coreos.com>
Date: Mon, 21 Jul 2014 16:11:00 -0700
Subject: [PATCH 33/37] networkd: fix reporting errors from hostnamed

The return value may be -EINVAL or a positive errno from the dbus
message. Check both ranges, otherwise most errors are silently ignored.
---
 src/network/networkd-link.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/network/networkd-link.c b/src/network/networkd-link.c
index a806990..24b4ebc 100644
--- a/src/network/networkd-link.c
+++ b/src/network/networkd-link.c
@@ -846,7 +846,9 @@ static int set_hostname_handler(sd_bus *bus, sd_bus_message *m, void *userdata,
 
         r = sd_bus_message_get_errno(m);
         if (r < 0)
-                log_warning_link(link, "Could not set hostname: %s", strerror(-r));
+                r = -r;
+        if (r > 0)
+                log_warning_link(link, "Could not set hostname: %s", strerror(r));
 
         return 1;
 }
-- 
1.8.5.5

